### PR TITLE
Create ReverseLinkedList.cs

### PR DESCRIPTION
**Pull request by rokovalenko**
Runtime: 87 ms, faster than 93.30% of C# online submissions for Reverse Linked List.
Memory Usage: 39.1 MB, less than 7.26% of C# online submissions for Reverse Linked List.

### CODE REVIEW

**Comment by rokovalenko** [comment]
I'm kind of surprised that my solution is faster than the default one (with the pointers).
I had no time this week and it felt like the easiest solution 😩

File: rkovalenko/ReverseLinkedList.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * public class ListNode {
+ *     public int val;
+ *     public ListNode next;
+ *     public ListNode(int val=0, ListNode next=null) {
+ *         this.val = val;
+ *         this.next = next;
+ *     }
+ * }
+ */
+public class Solution {
```
Comment (makcgr) [inline rkovalenko/ReverseLinkedList.cs:12]: Interesting take on the problem - another way to solve!

**Comment by vlemish** [comment]
> I'm kind of surprised that my solution is faster than the default one (with the pointers). I had no time this week and it felt like the easiest solution 😩

wow, nice job! looks like using additional memory always gives better performance in leetcode :D

File: rkovalenko/ReverseLinkedList.cs (hunk 1 above)
Comment (PShchegolevatykh) [reply rkovalenko/ReverseLinkedList.cs:12]: Yep, without links manipulation required two passes on a list and some additional memory. Good example.

**Review by PShchegolevatykh — APPROVED** [review]

**Pull request merged**